gum/backend-darwin/gumobjcapiresolver.c

```diff
@@ -171,38 +171,58 @@ gum_objc_api_resolver_enumerate_matches (GumApiResolver * resolver,
   GumObjcApiResolver * self = GUM_OBJC_API_RESOLVER (resolver);
   GMatchInfo * query_info;
   gchar method_type;
-  GPatternSpec * class_spec, * method_spec;
+  GPatternSpec  * modifiers_spec, * class_spec, * method_spec;
+  gboolean ignore_case = FALSE;
   GHashTableIter iter;
   gboolean carry_on;
   GHashTable * visited_classes;
   GumObjcClassMetadata * klass;
-
+  
   g_regex_match (self->query_pattern, query, 0, &query_info);
   if (!g_match_info_matches (query_info))
     goto invalid_query;
 
   method_type = gum_method_type_from_match_info (query_info, 1);
-  class_spec = gum_pattern_spec_from_match_info (query_info, 2);
-  method_spec = gum_pattern_spec_from_match_info (query_info, 3);
+  
+  modifiers_spec = gum_pattern_spec_from_match_info (query_info, 4, ignore_case);
+  if (modifiers_spec)
+  {
+    if (g_pattern_match_string (modifiers_spec, "i"))
+      ignore_case = TRUE;
+  }
 
+  class_spec = gum_pattern_spec_from_match_info (query_info, 2, ignore_case);
+  method_spec = gum_pattern_spec_from_match_info (query_info, 3, ignore_case);
+  
   g_match_info_free (query_info);
 
   g_hash_table_iter_init (&iter, self->class_by_handle);
   carry_on = TRUE;
   visited_classes = g_hash_table_new (NULL, NULL);
   while (carry_on && g_hash_table_iter_next (&iter, NULL, (gpointer *) &klass))
   {
-    if (g_pattern_match_string (class_spec, klass->name))
+    gchar * klass_name_copy = NULL;
+    if (ignore_case) 
+    {
+      klass_name_copy = g_ascii_strdown (klass->name, strlen (klass->name));
+    } 
+    else 
+    {
+      klass_name_copy = g_strdup (klass->name);
+    }
```

Comment:
```suggestion
    const gchar * class_name = klass->name;
    gchar * class_name_copy = NULL;

    if (ignore_case)
    {
      class_name_copy = g_ascii_strdown (class_name, -1);
      class_name = class_name_copy;
    }
```

- So we don't make a copy unless necessary.
- `g_ascii_strdown(len=-1)` should be used if the string is nul-terminated.
- Remove trailing whitespace.